### PR TITLE
Documentation updated

### LINKED ISSUE #2534: Documenting web3.eth.getChainId()
### Description 

The documentation for the JSON-RPC method ``eth_chainId`` on the Eth module is missing. 

Edit: 
Don't forget to update the types too.

#### Expected behavior

The Web3.js documentation does contain the method ``web3.eth.getChainId()``.

#### Actual behavior

The Web3.js documentation does not contain the method ``web3.eth.getChainId()``.

#### Steps to reproduce the behavior

1. Check the Eth module documentation

#### Error Logs 
-

### Versions 
-

### PR DESCRIPTION
**Pull request by nivida**
## Description

The module constructors where not consistently described.

Fixes #2534 #2514 #2494 #2490 #2383 

## Type of change

- [x] Documentation

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no warnings.
- [x] I have updated or added types for all modules I've changed
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` in the root folder with success and extended the tests if necessary.
- [x] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [x] I ran ```npm run dtslint``` in the root folder and tested that all my types are correct
- [x] I have tested my code on an ethereum test network.

### CODE REVIEW

File: docs/callbacks-promises-events.rst [hunk 1]
```diff
@@ -12,11 +12,11 @@ we provide multiple ways to act on asynchronous functions.
 Most web3.js objects allow a callback as the last parameter, as well as returning promises to chain functions.
 
 Ethereum as a blockchain has different levels of finality and therefore needs to return multiple "stages" of an action.
-To cope with requirement we return a "promiEvent" for functions like ``web3.eth.sendTransaction`` or contract methods.
-This "promiEvent" is a promise combined with an event emitter to allow acting on different stages of action on the blockchain, like a transaction.
+To cope with requirement we return a "PromiEvent" for functions like ``web3.eth.sendTransaction`` or contract methods.
+This "PromiEvent" is a promise combined with an event emitter to allow acting on different stages of action on the blockchain, like a transaction.
```
Comment (OFRBG) [inline docs/callbacks-promises-events.rst]: Maybe rephrase L15-L16 and L18-L19 as:

These stages are encapsulated into a "PromiEvent", which combines a promise with an event emitter. The event emitter fires an event for each of the finality stages.

An example of a function that benefits from PromiEvents is `web3.eth.sendTransaction`. The event emitter inside the promise fires events such as ``receipt`` and ``transactionHash``.

File: docs/getting-started.rst [hunk 2]
```diff
@@ -26,16 +25,29 @@ First you need to get web3.js into your project. This can be done using the foll
 
 - npm: ``npm install web3``
 - meteor: ``meteor add ethereum:web3``
-- pure js: link the ``dist/web3.min.js``
 
 After that you need to create a web3 instance and set a provider.
-Ethereum supported Browsers like Mist or MetaMask will have a ``ethereumProvider`` or ``web3.currentProvider`` available. For  web3.js, check ``Web3.givenProvider``.
-If this property is ``null`` you should connect to a remote/local node.
+Ethereum supported Browsers like Mist or MetaMask will have a ``ethereum`` or ``web3.currentProvider`` available.
+For  web3.js, check ``Web3.givenProvider``. If this property is ``null`` you should connect to your own local or remote node.
```
Comment (OFRBG) [inline docs/getting-started.rst:28]: For L30-L31, maybe make explicit MetaMask isn't a browser (yet), and that ethereum and currentprovider are window properties.

Browsers with Ethereum support, such as Mist and FireFox with Metamask, have the a window property called either ``ethereum`` or ``web3.currentProvider``.

With the web3.js library you can retrieve the browser's provider with ``Web3.givenProvider``. In case this property is ``null``, you need to connect to a node, local or remote.

File: docs/web3-eth.rst [hunk 3]
```diff
@@ -144,7 +144,7 @@ Property
 --------
 
 
-``String`` - 20 Bytes: Any ethereum address. You should have the private key for that address in your node or keystore. (Default is ``undefined``)
+``String`` - 20 Bytes: Any Ethereum address. You should have the private key for that address in your node or keystore. (Default is ``undefined``)
```
Comment (OFRBG) [inline docs/web3-eth.rst]: Use "need":

``String`` - 20 Bytes: Any Ethereum address. You need to have the private key for that address in your node or keystore. (Default is ``undefined``)

File: docs/web3-eth.rst [hunk 4]
```diff
@@ -419,8 +419,8 @@ getAccounts
 
     web3.eth.getAccounts([callback])
 
-Returns a list of accounts the node controls by using the provider and calling the RPC method ``eth_accounts``.
-If there are unlocked local accounts then it will return them instead of sending a request to the node.
+Will return a list of locally unlocked accounts or it will return the unlocked accounts from the currently connected node.
```
Comment (OFRBG) [inline docs/web3-eth.rst]: For phrasing consistency, start with a verb. The difference between "locally [connected node]" "currently connected node" isn't clear.

Returns a list of the locally unlocked accounts, or if connected remotely, the exposed and unlocked accounts of the provider.

File: docs/web3-eth.rst [hunk 5]
```diff
@@ -1168,17 +1161,18 @@ signTransaction
 
 .. code-block:: javascript
 
-    web3.eth.signTransaction(transactionObject [, callback])
+    web3.eth.signTransaction(transactionObject [, address,] [, callback])
 
 The method ``signTransaction`` signs a transaction with the private key of the given address.
-This method does only work if you're connected to a Parity node.
+It will sign the transaction locally if a local unlocked account with the given address is existing.
```
Comment (OFRBG) [inline docs/web3-eth.rst]: For L1166-L1167, rephrase to start with the verb describing the return. In context we already know it is ``signTransaction``. Should also include a clarification of what happens when the given address is *not* a local unlocked account.

Signs a transaction with the private key of the given ``from`` address.
If the given address is a local unlocked account, the transaction will be signed locally.

**Review by OFRBG — COMMENTED** [review]
Good changes! Keep in mind that users reading the docs don't know how the library behaves for "else" cases. If you specify a condition such as "if the address isn't available locally, X happens", the opposite needs to be implicitly clear or specified. Example logic: "[for all cases, Y happens], [only] if the address isn't available locally, X happens."

File: docs/web3-bzz.rst [hunk 6]
```diff
@@ -18,15 +18,15 @@ For more see the `Swarm Docs <http://swarm-guide.readthedocs.io/en/latest/>`_.
     import Web3 from 'web3';
     import {Bzz} from 'web3-bzz';
 
-    // will autodetect if the "ethereum" object is present and will either connect to the local swarm node, or the swarm-gateways.net.
+    // will autodetect if an provider is present and will either connect to the local swarm node, or the swarm-gateways.net.
```
Comment (OFRBG) [inline docs/web3-bzz.rst]: Should read "a provider".

File: docs/web3-eth-accounts.rst [hunk 7]
```diff
@@ -8,19 +8,15 @@ web3.eth.accounts
 
 The ``web3.eth.accounts`` contains functions to generate Ethereum accounts and sign transactions and data.
 
-.. note:: This package has NOT been audited and might potentially be unsafe. Take precautions to clear memory properly, store the private keys safely, and test transaction receiving and sending functionality properly before using in production!
-
-To use this package standalone use:
-
+.. note:: This package got NOT audited until now. Take precautions to clear memory properly, store the private keys safely, and test transaction receiving and sending functionality properly before using in production!
```
Comment (OFRBG) [inline docs/web3-eth-accounts.rst:11]: This is crucial. If you mean that the package is NOT audited, use the phrasing below. The current phrasing implies that it HAS been audited!

This package has not been audited. Take precautions to clear memory properly, store the private keys safely, and test transaction receiving and sending functionality properly before using in production!

File: docs/web3-eth-contract.rst [hunk 8]
```diff
@@ -72,8 +79,8 @@ Example
 .. code-block:: javascript
 
     const myContract = new web3.eth.Contract([...], '0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe', {
-        from: '0x1234567890123456789012345678901234567891', // default from address
-        gasPrice: '20000000000' // default gas price in wei, 20 gwei in this case
+        defaultAccount: '0x1234567890123456789012345678901234567891', // default from address
```
Comment (princesinha19) [inline docs/web3-eth-contract.rst:80]: I think you should change the comment for defaultAcount.

It should be the **`default account address`** instead of **default from address**

Comment (nivida) [reply docs/web3-eth-contract.rst:80]: I think it's more understandable with ``// default from address``. Because it's in the end the from property of the transaction.

Comment (princesinha19) [reply docs/web3-eth-contract.rst:80]: okay cool.
